pkg/kubelet/kubelet.go

```diff
@@ -698,6 +691,10 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	klet.streamingRuntime = runtime
 	klet.runner = runtime
 
+	// Period after which the runtime cache expires - set to slightly longer than
+	// the expected length between housekeeping periods, which explicitly refreshes
+	// the cache.
+	runtimeCacheRefreshPeriod := kubeCfg.PodHousekeepingPeriod.Duration + housekeepingWarningDuration
```

Comment:
> set to slightly longer than the expected length between housekeeping periods

this value seems to dependent on the actual value of PodHousekeepingPeriod, if we allow users to modify it, it seems we need to be able to calculate it in function of that, or just allow users to set both, but add validation for these dependant fields

Reply:
IIUC, the `slightly longer` means `housekeepingWarningDuration`, which is 1s. I don't see this 1s needs to be dependent on PodHousekeepingPeriod. Are you suggesting, we should, for example, make it 2s if PodHousekeepingPeriod is 200s?

Reply:
I don't know, I just read the comment but don't understand this logic, so I don't know if that second is an absolute or relative value

Reply:
If the user is updating this value, does it need [much] validation?  They will adjust and (hopefully) measure performance/impacts.